### ISSUE TITLE
training problem

### ISSUE THREAD

**Issue by CrossLee1** [pranavgade20/glide_text2im_train#1]
Thanks for providing the training code for GLIDE.
Did you succeed to reproduce the results?

**Comment by pranavgade20** [comment]
I didn't run the training on the complete dataset because I don't have the infrastructure to do that, but I did train it on 4 images and it seemed to overfit a lot and learn a bit.

If you do train it, please let me know how it goes!

**Comment by CrossLee1** [comment]
I am trying to train it using a small batch on COCO dataset.
A few questions:
1. seem not use the `text_cxt` argument, with variable length text, training may have problem
2. in https://github.com/pranavgade20/glide_text2im_train/blob/master/glide_text2im/text2im_model.py#L102, you use `mask` with all 1.
why not use `mask` by `padded_tokens_and_mask` in glide_text2im/tokenizer/bpe.py.

**Comment by pranavgade20** [comment]
1. I'm not sure what you mean, could you elaborate? The training code is mostly from https://github.com/openai/guided-diffusion which might be helpful to look at.
2. I'm very confused about why that's there at all as well! I think it was a hotfix to get it to work that I forgot to change later. I think it'll be an easy fix though, so feel free to create a PR if you do fix it at some point:)

**Comment by CrossLee1** [comment]
for question 1:
in https://github.com/pranavgade20/glide_text2im_train/blob/master/guided_diffusion/image_datasets.py#L134,
the length of `out_dict["tokens"]` will vary due to the different length of text prompt.
I think `text_ctx=128` shoule be used here to pad the token length to the same 128-dim.

**Comment by pranavgade20** [comment]
That makes sense. However, I don't remember encountering any issues because iirc it only uses one token and broadcasts it? Not completely sure about this, but I'm pretty sure padding isn't necessary here, but it should work with padding as well.

**Comment by CrossLee1** [comment]
Thanks for your reply.
Will continue to train, and it may take a long time to get some good results on my poor GPU :)

**Comment by pranavgade20** [comment]
Do let me know how it goes. I didn't get it to train on my GPU, so I'm definitely curious about your results!
